extended/src/main/java/io/kubernetes/client/extended/generic/GenericKubernetesApi.java

```diff
@@ -0,0 +1,680 @@
+package io.kubernetes.client.extended.generic;
+
+import com.google.common.base.Strings;
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.reflect.TypeToken;
+import io.kubernetes.client.custom.V1Patch;
+import io.kubernetes.client.extended.generic.options.*;
+import io.kubernetes.client.openapi.ApiClient;
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.Configuration;
+import io.kubernetes.client.openapi.apis.CustomObjectsApi;
+import io.kubernetes.client.openapi.models.V1DeleteOptions;
+import io.kubernetes.client.openapi.models.V1ObjectMeta;
+import io.kubernetes.client.openapi.models.V1Status;
+import io.kubernetes.client.util.PatchUtils;
+import io.kubernetes.client.util.Reflect;
+import io.kubernetes.client.util.Watch;
+import io.kubernetes.client.util.Watchable;
+import io.kubernetes.client.util.exception.ObjectMetaReflectException;
+import java.net.SocketTimeoutException;
+import okhttp3.Call;
+
+/**
+ * The Generic kubernetes api provides a unified client interface for not only the non-core-group
+ * built-in resources from kuberntes but also the custom-resources models meet the following
+ * requirements:
+ *
+ * <p>1. there's a `V1ObjectMeta` field in the model along with its getter/setter. 2. there's a
+ * `V1ListMeta` field in the list model along with its getter/setter. - supports Gson
+ * serialization/deserialization. 3. the generic kubernetes api covers all the basic operations over
+ * the custom resources including {get, list, watch, create, update, patch, delete}.
+ *
+ * @param <ApiType> the api type parameter
+ * @param <ApiListType> the api list type parameter
+ */
+public class GenericKubernetesApi<ApiType, ApiListType> {
+
+  // TODO(yue9944882): supports status operations..
+  // TODO(yue9944882): supports generic sub-resoruce operations..
+  // TODO(yue9944882): supports delete-collections..
+
+  private Class<ApiType> apiTypeClass;
+  private Class<ApiListType> apiListTypeClass;
+  private String apiGroup;
```

Comment:
Does it make sense to use the native Kubernetes object here for this? It has these fields.

Reply:
are you refering to [V1APIGroup](https://github.com/kubernetes-client/java/blob/master/kubernetes/src/main/java/io/kubernetes/client/openapi/models/V1APIGroup.java)? the so-called field "apiGroup" here should be the name of the serving resource api group e.g. `apps`, `networking.k8s.io`, `apiextensions.k8s.io`, etc. while the `V1APIGroup` is used for representing all the fundamental information about the an api group . we need only a tiny bit of that, which is V1APIGroup#name. simply a string value would be sufficient, wdyt?